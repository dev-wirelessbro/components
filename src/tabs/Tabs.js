import React from 'react'
import className from 'classnames'
import PropTypes from 'prop-types'
import TabPane from './TabPane'


export default class Tabs extends React.Component {

    static TabPane = TabPane

    static propTypes = {
        activeKey: PropTypes.string,
        onChange: PropTypes.func
    }

    constructor(props) {
        super(props)
        this.handleClick = this.handleClick.bind(this)
    }

    handleClick(activeKey) {
        this.props.onChange && this.props.onChange(activeKey)
    }

    buildTab() {
        const {activeKey} = this.props
        return React.Children.map(this.props.children, child => {
            const {key, props: {tab} } = child
            let className = 'nav-link'

            if (key === activeKey) {
                className = `${className} active`
            } else {
                className = `${className} text-primary`
            }

            return (
                <li key={key} className="nav-item" onClick={e => this.handleClick(key)}>
                    <span className={className}>{tab}</span>
                </li>
            )
        })
    }

    buildContent() {
        const {activeKey} = this.props
        return React.Children.map(this.props.children, child => {
            const {key, props: {children} } = child
            return key === activeKey ? (
                <div style={{marginTop: 10}}>
                    {children}
                </div>
            ): null
        })
    }

    render() {
        return (
            <div>
                <ul className="nav nav-tabs">
                    {this.buildTab()}
                </ul>
                {this.buildContent()}
            </div>
        )
    }
}
