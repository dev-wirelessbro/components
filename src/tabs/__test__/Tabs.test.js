import React from 'react'
import renderer from 'react-test-renderer'
import {shallow} from 'enzyme'
import Tabs from '../Tabs'


it('should match its snapshot without props', () => {
    const tree = renderer
    .create(<Tabs/>)
    .toJSON()
    expect(tree).toMatchSnapshot()
})

it('should match its snapshot with selectedKey only', () => {
    const tree = renderer
    .create(
        <Tabs activeKey='tab2'>
            <Tabs.TabPane key='tab2' tab='tab2'>
                tab2 content
            </Tabs.TabPane>
        </Tabs>
    )
    .toJSON()
    expect(tree).toMatchSnapshot()
})

it('should call onchange function', () => {
    // render the initial tabs
    const onChangeMock = jest.fn()
    let component = shallow(
        <Tabs activeKey='tab1' onChange={onChangeMock}>
            <Tabs.TabPane key='tab1' tab='tab1'>
                tab1 content
            </Tabs.TabPane>
            <Tabs.TabPane key='tab2' tab='tab2'>
                tab2 content
            </Tabs.TabPane>
        </Tabs>
    )
    // Find the second tab which is not selected
    const li = component.find('span').not('.active').parent().simulate('click')
    //
    expect(onChangeMock.mock.calls.length).toBe(1);
    expect(onChangeMock.mock.calls[0][0]).toBe('tab2');
})
