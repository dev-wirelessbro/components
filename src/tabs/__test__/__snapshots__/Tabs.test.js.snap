// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match its snapshot with selectedKey only 1`] = `
<div>
  <ul
    className="nav nav-tabs"
  >
    <li
      className="nav-item"
      onClick={[Function]}
    >
      <span
        className="nav-link active"
      >
        tab2
      </span>
    </li>
  </ul>
  <div
    style={
      Object {
        "marginTop": 10,
      }
    }
  >
    tab2 content
  </div>
</div>
`;

exports[`should match its snapshot without props 1`] = `
<div>
  <ul
    className="nav nav-tabs"
  />
</div>
`;
